lib/elixir/lib/process.ex

```diff
@@ -369,19 +369,35 @@ defmodule Process do
 
   See http://www.erlang.org/doc/man/erlang.html#process_info-2 for more info.
   """
-  @spec info(pid, atom) :: {atom, term} | nil
+  @spec info(pid, atom | [atom]) :: {atom, term} | [{atom, term}]  | nil
   def info(pid, spec)
 
   def info(pid, :registered_name) do
-    case :erlang.process_info(pid, :registered_name) do
+    handle_registered_name :erlang.process_info(pid, :registered_name)
+  end
+
+  def info(pid, spec) when is_atom(spec) or is_list(spec) do
```

Comment:
Oh GH comments, it's discussed in another comments branch: `Process.info(pid, [:registered_name])` needs special handling. :smile: